octane/evmengine/keeper/keeper.go

```diff
@@ -270,3 +281,21 @@ func (k *Keeper) getOptimisticPayload() (engine.PayloadID, uint64, time.Time) {
 
 	return k.mutablePayload.ID, k.mutablePayload.Height, k.mutablePayload.UpdatedAt
 }
+
+// withdrawalsEqual returns true only if two non-nil slices contain identical
+// withdrawals in identical order.
+func withdrawalsEqual(w1, w2 []*etypes.Withdrawal) bool {
+	if w1 == nil || w2 == nil || len(w1) != len(w2) {
```

Comment:
in general don't check slices for nil, always use len, since nil slice is equivalent to empty slice. Only check for nils of elements of a slice, not the slice itself.